templates/singleton/boil_queries.tpl

```diff
@@ -51,49 +51,85 @@ func mergeModels(tx boil.Executor, primaryID uint64, secondaryID uint64, foreign
 func deleteConflictsBeforeMerge(tx boil.Executor, conflict conflictingUniqueKey, primaryID uint64, secondaryID uint64) error {
 	conflictingColumns := strmangle.SetComplement(conflict.columns, []string{conflict.objectIdColumn})
 
-	if len(conflictingColumns) < 1 {
-		return nil
-	} else if len(conflictingColumns) > 1 {
-		return errors.Err("this doesnt work for unique keys with more than two columns (yet)")
+	selectParams := []interface{}{primaryID, secondaryID}
+	// 1 to 1 relationship check - highlander situation, there can be only one!
+	isFKManyto1 := 1
+	//Check that the compliment differs from the conflicting columns and that there is only 1 conflicting column
+	// representing the 1To1 case.
+	if len(conflict.columns) == 1 && len(conflictingColumns) == 0 {
+		isFKManyto1 = 0
+		conflictingColumns = conflict.columns
+		selectParams = []interface{}{secondaryID}
 	}
-
+	//This query checks for conflicting rows based on the keys. There are two cases. The first is when we have a 1To1
+	// relationship with only the objectID as the conflicting column. We want to grab all rows between both primary and
+	// secondary objectIDs because there can only be one once they are merged. The second is ManyTo1, where we want to
+	// grab the keys other than the objectIDs that conflict so we can grab all those having more than 1 distinct
+	// objectID because these will be merged and will conflict.
 	query := fmt.Sprintf(
-		"SELECT %s FROM %s WHERE %s IN (%s) GROUP BY %s HAVING count(distinct %s) > 1",
-		conflictingColumns[0], conflict.table, conflict.objectIdColumn,
-		strmangle.Placeholders(dialect.IndexPlaceholders, 2, 1, 1),
-		conflictingColumns[0], conflict.objectIdColumn,
+		"SELECT %s FROM %s WHERE %s IN (%s) GROUP BY %s HAVING count(distinct %s) > %d",
```

Comment:
would wrapping %s in \` help? I'm thinking about those column names that might have spaces or special keywords

Reply:
We discussed. I think if we have a column name that has spaces, this would not be accepted by SQLBoiler, but also if this is the case we should just fix the column name because it should be snake_case.